lib/timers/promises.js

```diff
@@ -136,13 +131,29 @@ function setImmediate(value, options = kEmptyObject) {
 }
 
 async function* setInterval(after, value, options = kEmptyObject) {
-  validateObject(options, 'options');
+  try {
+    if (typeof after !== 'undefined') {
+      validateNumber(after, 'delay');
+    }
+
+    validateObject(options, 'options');
+
+    if (typeof options?.signal !== 'undefined') {
+      validateAbortSignal(options.signal, 'options.signal');
+    }
+
+    if (typeof options?.ref !== 'undefined') {
+      validateBoolean(options.ref, 'options.ref');
+    }
+  } catch (err) {
+    return PromiseReject(err);
+  }
```

Comment:
Why catch the error in the first place? Isn't returning a promise rejection strictly equivalent in an async function?

Reply:
Yes, I copied and pasted from `setTimeout` (which is not async) and missed that. Do you think we can leave like that or do we need a tiny fixing PR?

Reply:
It would have been more "clean" to close that PR about adding an alias for `setTimeout`, and open a new one for the validation changes. That way, folks watching the repo would have had more chance to notice the change, and our wait time policy would have give enough time for folks to chime in (all the changes happen during a weekend, making it even more likely fewer folks had time to have a look). All in all, I don't really care if you do the follow-up PR with my suggestion, but IMO it's the "right" thing to do.